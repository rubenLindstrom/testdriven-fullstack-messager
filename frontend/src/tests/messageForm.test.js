import React from "react";
import MessageForm from "../components/messageForm";

import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("MessageForm", () => {
  it("renders without crashing", () => {
    const component = mount(<MessageForm />);
    expect(component).toMatchSnapshot();
  });

  it("has textbox", () => {
    const component = mount(<MessageForm />);
    expect(component.exists("textarea#message_box")).toBe(true);
  });

  it("has submit button", () => {
    const component = mount(<MessageForm />);
    expect(component.exists("button#submit")).toBe(true);
  });

  it("should update state message when tesxt entered", () => {
    const component = shallow(<MessageForm />);
    component
      .find("textarea#message_box")
      .simulate("change", { target: { value: "Hello" } });
    expect(component.state("currentMessage")).toEqual("Hello");
  });

  it("clears message box on submit", () => {
    const component = mount(<MessageForm onSubmit={(item) => true} />);
    component
      .find("textarea#message_box")
      .simulate("change", { target: { value: "Hello" } });
    expect(component.state("currentMessage")).toEqual("Hello");
    component.find("form").simulate("submit");

    expect(component.find("textarea#message_box").props().value).toEqual("");
    expect(component.state("currentMessage")).toEqual("");
  });
});
