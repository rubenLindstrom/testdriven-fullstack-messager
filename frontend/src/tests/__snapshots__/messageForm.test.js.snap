// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm renders without crashing 1`] = `ReactWrapper {}`;
